pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -376,14 +377,19 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 			continue
 		}
 		// Write the log line into the stream.
-		if err := writer.write(msg); err != nil {
+		if err := writer.write(msg, isNewLine); err != nil {
 			if err == errMaximumWrite {
 				klog.V(2).Infof("Finish parsing log file %q, hit bytes limit %d(bytes)", path, opts.bytes)
 				return nil
 			}
 			klog.Errorf("Failed with err %v when writing log for log file %q: %+v", err, path, msg)
 			return err
 		}
+		if msg.log[len(msg.log) - 1] == eol[0] {
+			isNewLine = true
+		} else {
+			isNewLine = false /* The subsequent content is not a newline */
```

Comment:
i'd write this as:
```
isNewLine = msg.log[len(msg.log) - 1] == eol[0]
```
and have a comment on top of the line explaining with a couple of sentences what this line does.